vembrane/__init__.py

```diff
@@ -162,12 +172,18 @@ def filter_vcf(
         env["FORMAT"] = Format(idx, formats)
         env["SAMPLES"] = list(record.samples)
 
-        annotations = dict(record.info).get(ann_key, [""])
-        filtered_annotations = [
-            annotation
-            for annotation in annotations
-            if eval_expression(expression, idx, annotation, annotation_keys, env,)
-        ]
+        if expression.filters_annotations():
+            annotations = dict(record.info).get(ann_key, [""])
+            filtered_annotations = [
+                annotation
+                for annotation in annotations
+                if expression.evaluate(idx, annotation, annotation_keys, env,)
+            ]
+        else:
+            annotations = [""]
+            filtered_annotations = (
+                [""] if expression.evaluate(idx, "", annotation_keys, env,) else []
+            )
```

Comment:
if there's no usage of ann_key in the expression, we do not need to re-evaluate the expression for every single annotation in the record but only need to do that exactly once